packages/cli/src/util/build/write-build-result.ts

```diff
@@ -0,0 +1,373 @@
+        OUTPUT_DIR,
+        'functions',
+        `${path}.prerender-config.json`
+      );
+      const prerenderConfig = {
+        ...output,
+        lambda: undefined,
+        fallback,
+      };
+      await fs.writeJSON(prerenderConfigPath, prerenderConfig, { spaces: 2 });
+    } else if (isFile(output)) {
+      await writeStaticFile(output, path, overrides, cleanUrls);
+    } else if (isEdgeFunction(output)) {
+      await writeEdgeFunction(output, path);
+    } else {
+      throw new Error(
+        `Unsupported output type: "${(output as any).type}" for ${path}`
+      );
+    }
+  }
+  return Object.keys(overrides).length > 0 ? overrides : undefined;
+}
+
+/**
+ * Writes the output from the `build()` return value of a v3 Builder to
+ * the filesystem.
+ */
+async function writeBuildResultV3(buildResult: BuildResultV3, build: Builder) {
+  const { output } = buildResult;
+  if (isLambda(output)) {
+    const src = build.src!;
+    const ext = extname(src);
+    const path = build.config?.zeroConfig
+      ? src.substring(0, src.length - ext.length)
+      : src;
+    await writeLambda(output, path);
+  } else {
+    throw new Error(
+      `Unsupported output type: "${(output as any).type}" for ${build.src}`
+    );
+  }
+}
+
+/**
+ * Writes a static `File` instance to the file system in the "static" directory.
+ * If the filename does not have a file extension then one attempts to be inferred
+ * from the extension of the `fsPath`.
+ *
+ * @param file The `File` instance to write
+ * @param path The URL path where the `File` can be accessed from
+ * @param overrides Record of override configuration when a File is renamed or has other metadata
+ */
+async function writeStaticFile(
+  file: File,
+  path: string,
+  overrides: Record<string, PathOverride>,
+  cleanUrls = false
+) {
+  let fsPath = path;
+  let override: PathOverride | null = null;
+
+  // If the output path doesn't match the determined file extension of
+  // the File then add the extension. This is to help avoid conflicts
+  // where an output path matches a directory name of another output path
+  // (i.e. `blog` -> `blog.html` and `blog/hello` -> `blog/hello.html`)
+  const ext = getFileExtension(file);
+  if (ext && extname(path) !== ext) {
+    fsPath += ext;
+    if (!override) override = {};
+    override.path = path;
+  }
+
+  // If `cleanUrls` is true then remove the `.html` file extension
+  // for HTML files.
+  if (cleanUrls && path.endsWith('.html')) {
+    if (!override) override = {};
+    override.path = path.slice(0, -5);
+  }
+
+  // Ensure an explicit "content-type" on the `File` is returned in
+  // the final output asset.
+  if (file.contentType) {
+    if (!override) override = {};
+    override.contentType = file.contentType;
+  }
+
+  if (override) {
+    overrides[fsPath] = override;
+  }
+
+  const dest = join(OUTPUT_DIR, 'static', fsPath);
+  await fs.mkdirp(dirname(dest));
+
+  // TODO: handle (or skip) symlinks?
```

Comment:
Should we make a card for this?

Reply:
https://linear.app/vercel/issue/BUI-122/handle-symlinks-when-writing-build-results